packages/roosterjs-content-model/lib/modelApi/common/mergeModel.ts

```diff
@@ -37,11 +38,14 @@ export interface MergeModelOption {
     insertPosition?: InsertPoint;
 
     /**
-     * When set to true, segment format of the insert position will be merged into the content that is merged into current model.
+     * Use this to decide whether to change the source model format when doing the merge.
+     * 'MergeCurrentFormat': segment format of the insert position will be merged into the content that is merged into current model.
      * If the source model already has some format, it will not be overwritten.
-     * @default false
+     * 'ApplyCurrentFormat': format of the insert position will be set into the content that is merged into current model.
+     * If the source model already has fontWeight, Italic or underline, it will not be overwritten.
+     * @default undefined
      */
-    mergeCurrentFormat?: boolean;
+    mergeFormatOption?: 'MergeCurrentFormat' | 'ApplyCurrentFormat';
```

Comment:
Better add a 'none' option to not merge. Although we can leave it undefined for "none", but in case the scenario that we already have an option object then we want to override it by assigning another option object to it and we want it not merge, then we can merge with "none" value.

Reply:
Also, for other cases of string literal values, we use strings start with lower cases. Although there is not a rule but I think we'd better make them consistent.

Reply:
also the names make me a little confusing.

In my understand, "merge" means both source and target format will be used, when there are conflict, we may have some rule to decide which one to use, here we may want to keep bold, italic, ...

And "apply" means only apply one side format to the other, if there are conflicts, always overwrite. 

So that is in a totally opposite way with your comment. Do you think we can switch them, or use some better words? I'm open for discussion.

Reply:
Also the property name `mergeFormatOption`, it is already inside an `MergeModelOption` interface, everything here is an option, so no need to add the word "option" in its name again.

Reply:
I agree with you on all. 

What about something like this:

```
    /**
     * Use this to decide whether to change the source model format when doing the merge.
     * 'targetAndSource': segment format of the insert position will be merged into the content that is merged into current model.
     * If the source model already has some format, it will not be overwritten.
     * 'currentFormatAndSourceEmphasis': format of the insert position will be set into the content that is merged into current model.
     * If the source model already has emphasis format, such as, fontWeight, Italic or underline different than the default style, it will not be overwritten.
     * 'none' No format merging will be applied to the source model.
     * @default undefined
     */
    mergeFormat: 'targetAndSource' | 'targetFormatAndSourceEmphasis' | 'none';
```

This way we would understand easier what type of merging we would do to the content.

Reply:
How about 'mergeAll' | 'keepSourceEmphasisFormat' | 'none'?

Reply:
Sounds good